### DISCUSSION TITLE
filter properties

### DISCUSSION THREAD

**Discussion by jkremolador** [gmtsar/gmtsar#719] [Q&A]
Hello,

I am trying to produce phase gradients from coseismic interferograms. However, I am not quite sure how to select the proper filter to use (e.g., choice of filter wavelength).

In my case for example, I have to work with S1 interferograms already multilooked by a factor of 2 in azimuth and 8 in range. For some reason, I have to work with multilooked instead of the full resolution interferograms. How would I know what filter wavelength is appropriate to use in such case? Do we just do trial-and-error until a desirable result is produced, or is there a more systematic way? 

Thanks for the clarification.

**Comment by Xiaohua-Eric-Xu** [comment]
For phase gradient on S1 TOPS data, try setting range_dec = 4, azimuth_dec = 1, filter_wavelength = 60 and compute_phase_gradient = 1.

**Reply by Xiaohua-Eric-Xu** [reply]
If you multi-look, the quality may degrade. Increase the first three parameters in a linear manner.

**Reply by jkremolador** [reply]
Can you explain how the 60 m wavelength for S1 TOPS was derived/chosen?

**Reply by Xiaohua-Eric-Xu** [reply]
Max azimutal pixel size is ~13m. For a guassian filter of 60m the eventual sampling at radar coordinates is about 15m.

**Comment by jkremolador** [comment]
Hi @Xiaohua-Eric-Xu we tried replicating the phase gradient computation with a python-based code that feeds in data from the French Space Agency's (CNES) FLATSIM NSBAS processing service (i.e., S1 interferograms in 2-looks resolution). We tested the computation on Ridgecrest coseismic interferograms and we have produced an initial result using the appropriate filter properties (wavelength=120), as shown below. We found it to be visually similar with your results from the Xu et al. (2020) paper.

![fig_phgr_results](https://github.com/gmtsar/gmtsar/assets/78633479/f041b579-e6c3-4f53-8fd9-8b48f7a6301c)
![fig_xu2020_results](https://github.com/gmtsar/gmtsar/assets/78633479/d57413d1-524e-4c68-9ee8-1062272b9883)

We were wondering if we could request for a copy of your phase gradient files on Ridgecrest. We found the files from the link indicated in the paper (https://topex.ucsd.edu/SV_7.1/index.html), but the phase gradient files were in kmz form and do not preserve the values in rad/pixel. We were hoping to have a copy of the original files so that we could compare them directly with our reproduced results. Thank you.

**Reply by Xiaohua-Eric-Xu** [reply]
Could you try this link here: https://drive.google.com/drive/folders/1JoxR7ivlXJixT0sHJOtFKkhHhoaSbJTz?usp=share_link

**Reply by Xiaohua-Eric-Xu** [reply]
I used 60m filter by the way. And the results are stacked so the noise level might be smaller as shown by this comparison below.
![image](https://github.com/gmtsar/gmtsar/assets/16844111/5d6252fd-73b3-4b91-a637-902efd043e30)

**Reply by jkremolador** [reply]
Great, thank you. I can access the link just fine. We used 120 m filter because we had to work with already multilooked interferograms.
